app/controllers/api/v3/search.rb

```diff
@@ -99,6 +99,36 @@ def forwarded_ip_address
           serialized_bikes_results(paginate(close_serials))
         end
 
+        desc "Search by serial number (partially matching)", {
+               notes: <<-NOTE,
+                This endpoint accepts the same parameters as the root `/search`
+                endpoint.
+
+                It returns bikes with partially-matching serial numbers to the
+                requested serial.
+
+                If space-separated, multiple serial number stems can be provided
+                under the `serial` param and partial matches for all will be
+                returned.
+               NOTE
+             }
+        paginate
+        params do
+          requires :serial, type: String, desc: "Serial, homoglyph matched"
+          use :non_serial_search_params
+          optional :per_page, type: Integer, default: 25, desc: "Bikes per page (max 100)"
+        end
+        get "/partial_serials" do
+          results =
+            params[:serial]
```

Comment:
I had to run this code in IRB to figure out what it was doing. Can you make this a little more straightforward?

Why not

```ruby
results = Bike.search_partial_serials(interpreted_params.merge(serial: params[:serial]))
```

Or even dropping the merge (I don't understand why it's necessary?) and just doing

```ruby
serialized_bikes_results(paginate(Bike.search_partial_serials(interpreted_params))))
```

Reply:
The split and merge is there to account for multiple serials requirement:

>  If space-separated, multiple serial number stems can be provided under the `serial` param and partial matches for all will be returned.

Let me know if the explaining variables in 429637e1 seem more intuitive.

I'm not sure how I feel about pushing the multiple-serial-handling down the stack into `Bike.search_partial_serials` — it would be a departure from the design of its sibling methods, and in isolation would either be unintuitive in the single-query case (in which it would need to accept an array of a single hash) or take some doing to not make unintuitive.

Reply:
Hmmm. Ok, looking at this now - I think it might actually be a frontend thing.

If your search has spaces, the frontend splits it into multiple blocks, so it shows:

WITH SERIAL NUMBERS PARTIALLY MATCHING AJ6O7L

and another block:

WITH SERIAL NUMBERS PARTIALLY MATCHING 888